src/helpers/git/get-git-data.ts

```diff
@@ -9,18 +9,28 @@ export const gitRemote = async (git: simpleGit.SimpleGit): Promise<string> => {
   if (remotes.length === 0) {
     throw new Error('No git remotes available')
   }
+  const defaultRemote = await getDefaultRemoteName(git)
 
   for (const remote of remotes) {
-    // We're trying to pick the remote called with the default git name 'origin'.
-    if (remote.name === 'origin') {
+    if (remote.name === defaultRemote) {
       return stripCredentials(remote.refs.push)
     }
   }
 
-  // Falling back to picking the first remote in the list if 'origin' is not found.
+  // Falling back to picking the first remote in the list if the default remote is not found.
   return stripCredentials(remotes[0].refs.push)
 }
 
+export const getDefaultRemoteName = async (git: simpleGit.SimpleGit): Promise<string> => {
+  try {
+    const defaultRemoteConfig = (await git.getConfig('clone.defaultRemoteName')) ?? {}
+
+    return defaultRemoteConfig.value ?? 'origin'
```

Comment:
nitpick: this is up to preference too, but you can do this in a single line: 
```suggestion
    return (await git.getConfig('clone.defaultRemoteName'))?.value ?? 'origin'
```

Reply:
Nice, I didn't know about that syntax!